query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverter.java

```diff
@@ -285,9 +296,9 @@ private String convertExpression2String(
   }
 
   private String convertExpressionToMapKeyColumn(Expression expression) {
-    if (expression.getValueCase() == COLUMNIDENTIFIER) {
-      String logicalColumnName = expression.getColumnIdentifier().getColumnName();
-      String col = viewDefinition.getKeyColumnNameForMap(logicalColumnName);
+    if ((expression.getValueCase() == COLUMNIDENTIFIER)
+        || (expression.getValueCase() == ATTRIBUTE_EXPRESSION && !isComplexAttribute(expression))) {
```

Comment:
nit: `(expression.getValueCase() == ATTRIBUTE_EXPRESSION && !isComplexAttribute(expression))` could be extracted into a method

Reply:
resolved